sys/include/net/netopt.h

```diff
@@ -269,6 +269,54 @@ typedef enum {
      */
     NETOPT_LAST_ED_LEVEL,
 
+    /** @brief   Radio modulation bandwidth
+     *
+     * Depends on the type of radio used and the regional regulatory and is
+     * used with LoRa/FSK modems.
+     *
+     * Actual implementations support 125kHz, 250kHz and 500kHz.
+     *
+     * @see drivers_sx127x
+     */
+    NETOPT_BANDWIDTH,
+
+    /** @brief   Radio spreading factor
+     *
+     * For LoRa/FSK modems, allowed values are in the range 7..12.
+     */
+    NETOPT_SPREADING_FACTOR,
+
+    /** @brief   Radio coding rate
+     *
+     * For LoRa/FSK modems, allowed values are in the range 5..8.
+     */
+    NETOPT_CODING_RATE,
+
+    /** @brief   Semtech module modem type.
+     *
+     * Allowed values are LoRa and FSK.
+     *
+     * FSK is not supported for the moment.
+     */
+    NETOPT_MODEM_TYPE,
+
+    /**
+     * @brief LoRA specific options.
+     */
+    NETOPT_LORA_SINGLE_RECEIVE,      /**< When enabled, RX is turned off upon reception of a packet */
+    NETOPT_LORA_RX_TIMEOUT,          /**< Reception timeout of a LoRa packet */
+    NETOPT_LORA_TX_TIMEOUT,          /**< Transmission timeout of a LoRa packet */
+    NETOPT_LORA_SYMBOL_TIMEOUT,      /**< Symbol timeout */
+    NETOPT_LORA_SYNCWORD,            /**< Syncword */
+    NETOPT_LORA_HOP,                 /**< Enable/disable frequency hopping */
+    NETOPT_LORA_HOP_PERIOD,          /**< Frequency hopping period */
+    NETOPT_CRC,                      /**< Enable/disable CRC check */
+    NETOPT_LORA_FIXED_HEADER,        /**< Enable/disable fixed header mode */
+    NETOPT_LORA_IQ_INVERT,           /**< Enable/disable IQ inverted */
+    NETOPT_LORA_PAYLOAD_LENGTH,      /**< Payload length */
```

Comment:
Actually, (at least in IEEE 802.15.4 and Ethernet) `NETOPT_MAX_PACKET_SIZE` returns the link-layer payload length (i.e. the maximum L3 packet length, wording was originally chosen since Frame == L2 PDU, Packet == L3 PDU).